docs/machine-learning/how-to-guides/train-model-textual-ml-net.md

```diff
@@ -0,0 +1,76 @@
+ms.date: 11/07/2018
+ms.topic: how-to
+ms.custom: mvc
+#Customer intent: As a developer, I want to apply feature engineering for my model training on textual data with ML.NET so that I can use my model in the ML.NET processing pipeline.
+---
+# Applying feature engineering for machine learning model training on textual data with ML.NET
+
+You need to convert any non float data to `float` data types since all ML.NET `learners` expect features as a `float vector`.
+
+To learn on textual data, you need to extract text features. ML.NET has some basic text feature extraction mechanisms:
+
+- `Text normalization` (removing punctuation, diacritics, switching to lowercase etc.)
+- `Separator-based tokenization`.
+- `Stopword` removal.
+- `Ngram` and `skip-gram` extraction.
+- `TF-IDF` rescaling.
+- `Bag of words` conversion.
+
+The following example demonstrates ML.NET text feature extraction mechanisms using the 
+[Wikipedia detox dataset](https://github.com/dotnet/machinelearning/blob/master/test/data/wikipedia-detox-250-line-data.tsv):
+
+```console
+Sentiment   SentimentText
+1	Stop trolling, zapatancas, calling me a liar merely demonstartes that you arer Zapatancas. You may choose to chase every legitimate editor from this site and ignore me but I am an editor with a record that isnt 99% trolling and therefore my wishes are not to be completely ignored by a sockpuppet like yourself. The consensus is overwhelmingly against you and your trolling lover Zapatancas,  
+1	::::: Why are you threatening me? I'm not being disruptive, its you who is being disruptive.   
+0	" *::Your POV and propaganda pushing is dully noted. However listing interesting facts in a netral and unacusitory tone is not POV. You seem to be confusing Censorship with POV monitoring. I see nothing POV expressed in the listing of intersting facts. If you want to contribute more facts or edit wording of the cited fact to make them sound more netral then go ahead. No need to CENSOR interesting factual information. "
+0	::::::::This is a gross exaggeration. Nobody is setting a kangaroo court. There was a simple addition concerning the airline. It is the only one disputed here.   
+```
+
+```csharp
+// Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+// as a catalog of available operations and as the source of randomness.
+var mlContext = new MLContext();
+
+// Define the reader: specify the data columns and where to find them in the text file.
+var reader = mlContext.Data.TextReader(ctx => (
+        IsToxic: ctx.LoadBool(0),
+        Message: ctx.LoadText(1)
+    ), hasHeader: true);
+
+// Read the data.
+var data = reader.Read(dataPath);
+
+// Inspect the message texts that are read from the file.
+var messageTexts = data.GetColumn(x => x.Message).Take(20).ToArray();
+
+// Apply various kinds of text operations supported by ML.NET.
+var learningPipeline = reader.MakeNewEstimator()
+    .Append(r => (
```

Comment:
Is something missing here? Does the `Append` method have a version that takes a lambda expression that returns a Tuple? (It may, I don't know. And because this is just pasted code it's hard to validate)

Reply:
This is an older version of the code. I will update it. Good catch